runtime/src/qio/qio_formatted.c

```diff
@@ -764,6 +803,248 @@ qioerr qio_channel_scan_string(const int threadsafe, qio_channel_t* restrict ch,
+  int32_t tmpchr;
+  int handle_back;
+  int handle_0x;
+  int stop_space;
+  unsigned long conv;
+  qio_style_t* style;
+  ssize_t nread = 0;
+  int64_t mark_offset;
+  int64_t end_offset;
+  ssize_t maxlen_chars = SSIZE_MAX - 1;
+  int found_term = 0;
+
+  if( maxlen_bytes <= 0 ) maxlen_bytes = SSIZE_MAX - 1;
+
+  if( threadsafe ) {
+    err = qio_lock(&ch->lock);
+    if( err ) return err;
+  }
+
+  style = &ch->style;
+
+  if( style->max_width_characters < UINT32_MAX &&
+      style->max_width_characters < maxlen_chars ) {
+    maxlen_chars = style->max_width_characters;
+  }
+  if( style->max_width_bytes < UINT32_MAX &&
+      style->max_width_bytes < maxlen_bytes ) {
+    maxlen_bytes = style->max_width_bytes;
+  }
+
+  // Allocate room for our buffer...
+  err = _append_byte(&ret, &ret_len, &ret_max, 0);
+  if( err ) goto unlock;
+  ret_len = 0;
+
+  mark_offset = qio_channel_offset_unlocked(ch);
+
+  err = qio_channel_mark(false, ch);
+  if( err ) goto unlock;
+
+  term_chr = style->string_end;
+  if( style->string_format == QIO_STRING_FORMAT_WORD ) {
+    handle_back = 0;
+    handle_0x = 0;
+    stop_space = 1;
+  } else if( style->string_format == QIO_STRING_FORMAT_BASIC ) {
+    handle_back = 1;
+    handle_0x = 0;
+    stop_space = 0;
+  } else if( style->string_format == QIO_STRING_FORMAT_CHPL ) {
+    handle_back = 1;
+    handle_0x = 1;
+    stop_space = 0;
+  } else if( style->string_format == QIO_STRING_FORMAT_JSON ) {
+    handle_back = 1;
+    handle_0x = 0;
+    stop_space = 0;
+  } else if( style->string_format == QIO_STRING_FORMAT_TOEND ) {
+    handle_back = 0;
+    handle_0x = 0;
+    stop_space = 0;
+  } else if( style->string_format == QIO_STRING_FORMAT_TOEOF ) {
+    handle_back = 0;
+    handle_0x = 0;
+    stop_space = 0;
+    term_chr = -1;
+  } else {
+    handle_back = 1;
+    handle_0x = 1;
+    stop_space = 0;
+  }
+
+  err = 0;
+  for( nread = 0;
+      // limit # characters
+      nread < maxlen_chars &&
+      // stop on error
+      !err &&
+      // limit # bytes
+      qio_channel_offset_unlocked(ch) - mark_offset < maxlen_bytes;
+      nread++ ) {
+    chr = qio_channel_read_byte(false, ch);
+    if(chr < 0) {
+      err = qio_int_to_err(-chr);
+      break;
+    }
+
+    // If we're using FORMAT_WORD, skip any whitespace at the beginning
+    if( nread == 0 ) {
+      while( !(style->string_format == QIO_STRING_FORMAT_TOEND ||
+               style->string_format == QIO_STRING_FORMAT_TOEOF) &&
+             iswspace(chr) ) {
+        // Read the next byte!
+        chr = qio_channel_read_byte(false, ch);
+        if(chr < 0) {
+          err = qio_int_to_err(-chr);
+          break;
+        }
+      }
+      if( err ) break;
+
+      if( style->string_format == QIO_STRING_FORMAT_WORD ||
+          style->string_format == QIO_STRING_FORMAT_TOEND ||
+          style->string_format == QIO_STRING_FORMAT_TOEOF ) {
+        // OK, use the byte we have
+      } else if( chr == style->bytes_prefix ) {
```

Comment:
I would expect that one use case would be to want to read a bytes without the b; that is formatted I/O that reads a string with "bla" into a bytes (with no `b"`). So I'd make this use style->bytes_prefix only if it is not 0.